cmp/form/FormModel.ts

```diff
@@ -274,4 +308,83 @@ export class FormModel extends HoistModel {
             }
         );
     }
+
+    private initPersist({
+        fieldsToInclude = null,
+        fieldsToExclude = null,
+        path = 'formValues',
+        ...rootPersistWith
+    }: FormPersistOptions) {
+        const fieldNameMap = this.createFormFieldsToPersistListing(
+            fieldsToInclude,
+            fieldsToExclude
+        );
+
+        PersistenceProvider.create({
+            persistOptions: {
+                path,
+                ...rootPersistWith
+            },
+            target: {
+                getPersistableState: () => {
+                    return new PersistableState(
```

Comment:
Since the `serialize` and `deserialize` methods are recursive, could simplify this to just:


```
getPersistableState: () =>
                    new PersistableState(this.serialize(pick(this.getData(), fieldNameMap))),


setPersistableState: ({value: formValues}) =>
                    this.setValues(this.deserialize(pick(formValues, fieldNameMap)))
```